### PR TITLE
Run tests with escript

### PR DESCRIPTION
**Pull request by kytrinyx**
This takes about a second as opposed to the old test runner that takes ~30 seconds.

Cherry-picked from #29

### CODE REVIEW

**Comment by kytrinyx** [comment]
All the tests pass, but the escript checker is exiting with status 31 instead of 0. I'm not sure where to begin figuring out why that's the case. @tmcgilchrist / @ebengt any ideas?

File: _test/check-exercises.escript [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#! /usr/bin/env escript
+
+main( [] ) -> 
+	Examples = filelib:wildcard( "*/example.erl" ),
+	Modules = [{X, compile(X)} || X <- Examples],
+	[compile_tests(X) || X <- Modules],
+	Results = [run_tests(X) || X <- Modules],
+	erlang:halt( erlang:length([X || X <- Results, X =:= ok]) );
```
Comment (tmcgilchrist) [inline _test/check-exercises.escript]: This is returning the number of passed tests, http://www.erlang.org/doc/man/erlang.html#halt-1
It should be checking for failures i.e.
```erlang
  erlang:halt( erlang:length([X || X <- Results, X =/= ok]) );
```

File: _test/check-exercises.escript [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+compile_tests( {Example, {Example_module, _Binary}} ) ->
+	Filename = erlang:atom_to_list(Example_module) ++ "_tests.erl",
+	Filepath = filename:join( [filename:dirname(Example), Filename] ),
+	compile( Filepath ).
+
+
+run_tests( {_Example, {Module, _Binary}} ) ->
+	io:fwrite( "~p: ", [Module] ),
+	eunit:test( Module ).
```
Comment (tmcgilchrist) [inline _test/check-exercises.escript]: Consider changing this to verbose `eunit:test( Module , [verbose]).`
http://www.erlang.org/doc/man/eunit.html#test-1

Verbose
```bash
...
  All 10 tests passed.
word_count: ======================== EUnit ========================
module 'word_count'
  module 'word_count_tests'
    word_count_tests: count_one_word_test...ok
    word_count_tests: count_one_of_each_test...ok
    word_count_tests: count_multiple_occurrences_test...ok
    word_count_tests: ignore_punctuation_test...ok
    word_count_tests: include_numbers_test...ok
    word_count_tests: normalize_case_test...ok
    word_count_tests: prefix_punctuation_test...ok
    word_count_tests: symbols_are_separators_test...ok
    [done in 0.023 s]
  [done in 0.023 s]
=======================================================
  All 8 tests passed.
```
vs 
```bash
...
word_count:   All 8 tests passed.
```

Comment (kytrinyx) [inline _test/check-exercises.escript]: Yeah, good call. I added that.

**Comment by tmcgilchrist** [comment]
Nice work :rocket:

**Pull request merged**